components/ws-manager-mk2/controllers/workspace_controller.go

```diff
@@ -157,7 +159,27 @@ func (r *WorkspaceReconciler) actOnStatus(ctx context.Context, workspace *worksp
 			} else {
 				// TODO(cw): replicate the startup mechanism where pods can fail to be scheduled,
 				//			 need to be deleted and re-created
-				workspace.Status.PodStarts++
+				// Must increment and persist the pod starts, and ensure we retry on conflict.
+				// If we fail to persist this value, it's possible that the Pod gets recreated
+				// when the workspace stops, due to PodStarts still being 0 when the original Pod
+				// disappears.
+				newStarts := workspace.Status.PodStarts + 1
+				if err = retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+					err := r.Get(ctx, types.NamespacedName{Name: workspace.Name, Namespace: workspace.Namespace}, workspace)
```

Comment:
If we get the workspace here we would overwrite any changes to the workspace we have made so far right? Should not be a problem because the only change we could have made at this point is setting the phase from "" to Pending which will be reconciled in the next loop. So not a problem, just pointing it out.

Reply:
Ah yeah good point. Not great, we could try and change this to a Patch to prevent conflicts and having to read the workspace from the API?

Reply:
Sounds good!

Reply:
Updated to use Patch, but realised we'd still have the same issue, as also on running `client.Patch()` k8s will save back the latest status returned by the api into the object.

So I've moved updating the status inside the main control loop to before `actOnStatus`, so that the updates from `updateWorkspaceStatus()` get saved, and then `actOnStatus` can patch without losing those updates. Wdyt?